### PR TITLE
chore: share `getHumanReadableProjectTypes` among email templates

### PR DESCRIPTION
**Pull request by jessicamcinchak**
Changes:
- Updates `getHumanReadableProjectTypes` utils method to accept a passport-like dictionary of key/values, rather than an entire session, allowing us to re-use it between Save & Return (by passing `lowcal_sessions.data.passport.data`) and Invite to Pay (by passing `payment_requests.session_preview_data`)
- Tidies up payment request emails & validation to reference new db fields `applicant_name` & `payment_amount`
- Updates test data & query mocks accordingly

Next up:
- Move Notify-related utils to their own subdirectory in the API

### CODE REVIEW

**Comment by jessicamcinchak** [comment]
[Improve API handling for configuring and sending GOV.UK Notify emails](https://trello.com/c/b1HtcRgM/2371-improve-api-handling-for-configuring-and-sending-govuk-notify-emails)

File: api.planx.uk/inviteToPay/sendPaymentEmail.ts [hunk 1]
```diff
@@ -8,8 +8,9 @@ const sendSinglePaymentEmail = async (
 ) => {
   try {
     const payment = await validatePaymentRequest(paymentRequestId, template);
-    if (!payment || Boolean(payment?.paid_at)) throw Error(`Invalid payment request: ${paymentRequestId}`);
-
+    if (!payment) throw Error("Unable to send payment email, cannot find payment request");
```
Comment (jessicamcinchak) [inline api.planx.uk/inviteToPay/sendPaymentEmail.ts:11]: this addresses lingering feedback from #1592, moving validation checks (eg is it already paid) to the `validatePaymentRequest` method below

File: api.planx.uk/saveAndReturn/utils.ts [hunk 2]
```diff
@@ -296,10 +296,10 @@ const markSessionAsSubmitted = async (sessionId: string) => {
  * Get formatted list of the session's project types
  */
 const getHumanReadableProjectType = async (
-  session: LowCalSession
+  sessionData: LowCalSession["data"]["passport"]["data"] | Record<string, any>
```
Comment (DafyddLlyr) [inline api.planx.uk/saveAndReturn/utils.ts:299]: Please see https://github.com/theopensystemslab/planx-new/pull/1633/files#r1169674642

This is a good choice 👍

**Review by DafyddLlyr — APPROVED** [review]
Apologies again for also covering some of this ground in https://github.com/theopensystemslab/planx-new/pull/1633 - just plain missed this PR.

Let's get this merged and I'll rebase and take it from there 😄

**Pull request merged**